d3a_interface/utils.py

```diff
@@ -132,5 +134,18 @@ def mkdir_from_str(directory: str, exist_ok=True, parents=True):
 class RepeatingTimer(Timer):
     def run(self):
         while not self.finished.is_set():
-            self.function(*self.args, **self.kwargs)
             self.finished.wait(self.interval)
+            self.function(*self.args, **self.kwargs)
+
+
+def check_redis_health(redis_db):
+    result = False
```

Comment:
If you can rename it to `is_connected` would be great, it would better reflect the intent here